### PR TITLE
Adds support for natural language use with IStringLocalizer<T>

### LINKED ISSUE #3: Failure on resource keys with non-alphanumeric characters
If the resx file key is not a valid C# member identifier, the generated code fails.
```xml
  <data name="{0} is not {1}" xml:space="preserve">
    <value>{0} is not {1}</value>
  </data>
```

generates
```CSharp
public static string? {0} is greater than {1} => ResourceManager.GetString(nameof({0} is greater than{1}), CultureInfo);
```

Corresponding generated Designer.cs code by VS uses underscores to replace non-identifier characters (and literal string in GetString):

```CSharp
public static string _0__is_greater_than__1_ => ResourceManager.GetString("{0} is greater than{1}", CultureInfo); 
```

This is a useful resource format especially for `IStringLocalizer<>` cases, e.g. using
```csharp
// e.g. from dependency injection:
IStringLocalizer<MyResource> strings;

this.ValidationError = strings["{0} is greater than {1}", 20, 10];
```

### PR DESCRIPTION
**Pull request by mickut**
This PR adds support for generating resource properties with keys that are not valid C# identifiers, enables creating non-static resource classes for use with `IStringLocalizer<T>` from `Microsoft.Extensions.Localization` (can't use static types in generics), and setting the resource namespace from `TargetPath` property of a linked `EmbeddedResource` to match the resource path.

Resolves #3

### CODE REVIEW

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs [hunk 1]
```diff
@@ -116,7 +116,9 @@ namespace Resources
 				CustomToolNamespace: "Resources",
 				ClassName: "ActivityEntrySortRuleNames",
 				PublicClass: publicClass,
-				NullForgivingOperators: false
+				NullForgivingOperators: false,
+        StaticClass: true
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Style nit: Convert indentation to tabs.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs [hunk 2]
```diff
@@ -163,7 +165,8 @@ namespace Resources
 				CustomToolNamespace: "Resources",
 				ClassName: "ActivityEntrySortRuleNames",
 				PublicClass: publicClass,
-				NullForgivingOperators: true
+				NullForgivingOperators: true,
+        StaticClass: true
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Style nit: Convert indentation to tabs.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs [hunk 3]
```diff
@@ -390,12 +393,123 @@ public static class CommonMessages
 				CustomToolNamespace: null,
 				ClassName: "CommonMessages",
 				PublicClass: true,
-				NullForgivingOperators: false
+				NullForgivingOperators: false,
+        StaticClass: true
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Style nit: Convert indentation to tabs.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs [hunk 4]
```diff
@@ -473,7 +587,8 @@ public void Generate_StringBuilder_Name_InvalidCharacter()
 			CustomToolNamespace: "Resources",
 			ClassName: "ActivityEntrySortRuleNames",
 			PublicClass: true,
-			NullForgivingOperators: false
+			NullForgivingOperators: false,
+      StaticClass: true
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Style nit: Convert indentation to tabs.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs [hunk 5]
```diff
@@ -555,7 +670,8 @@ public void Generate_StringBuilder_Value_InvalidCharacter()
...
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Style nit: Convert indentation to tabs.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs [hunk 6]
```diff
@@ -390,12 +393,123 @@ public static class CommonMessages
...
-				NullForgivingOperators: false
+				NullForgivingOperators: false,
+        StaticClass: true
 			)
 		);
 		source.Should().Be(expected);
 	}
 
+  [Fact]
+  public void Generate_StringBuilder_Name_NotValidIdentifier()
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Can you please create a folder something like `github-issues/3` and move this test into it?

File: VocaDb.ResXFileCodeGenerator/StringBuilderGenerator.cs [hunk 7]
```diff
@@ -32,7 +35,7 @@ public string Generate(Stream resxStream, GeneratorOptions options)
 
 		builder.Append("    ");
 		builder.Append(options.PublicClass ? "public" : "internal");
-		builder.Append(" static class ");
+		builder.Append(options.StaticClass ? " static class " : " class ");
```
Comment (ycanardeau) [inline VocaDb.ResXFileCodeGenerator/StringBuilderGenerator.cs:38]: Does it make sense to make not only classes, but also properties, non-static?

Comment (mickut) [reply VocaDb.ResXFileCodeGenerator/StringBuilderGenerator.cs:38]: The properties are static regardless, to enable "old-school" way of accessing with `MyResource.MyLocalizedString`.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs (hunk 1 above)
Comment (mickut) [reply VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Fixed. My VSCode was confused by the space-indented expected value strings and didn't follow .editorconfig for some reason.

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs (hunk 2 above)
Comment (mickut) [reply VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Fixed

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs (hunk 4 above)
Comment (mickut) [reply VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Fixed

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs (hunk 5 above)
Comment (mickut) [reply VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Fixed

File: VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs (hunk 6 above)
Comment (mickut) [reply VocaDb.ResXFileCodeGenerator.Tests/GeneratorTests.cs]: Refactored to GithubIssues/Issue3/GeneratorTests.cs for this test.

**Comment by mickut** [comment]
Ran into another problem when doing fixes on Mac. The newline expectations failed, fixed in code to use `EnvironmentNewLine` and the tests to use `string.ReplaceNewLines()` method on generated and expected values.

**Comment by ycanardeau** [comment]
Thanks for the PR! I've published a NuGet package here: https://www.nuget.org/packages/VocaDb.ResXFileCodeGenerator/2.0.2

I wish you a happy new year!

**Pull request merged**